packages/swing-store/src/swingStore.js

```diff
@@ -623,6 +835,53 @@ export function initSwingStore(dirPath = null, options = {}) {
   return makeSwingStore(dirPath, true, options);
 }
 
+/**
+ * @param {SwingStoreExporter} exporter
+ * @param {string | null} [dirPath]
+ * @param {object} options
+ * @returns {Promise<SwingStore>}
+ */
+export async function importSwingStore(exporter, dirPath = null, options = {}) {
+  if (dirPath) {
+    assert.typeof(dirPath, 'string');
+  }
+  const { includeHistorical = false } = options;
+  const store = makeSwingStore(dirPath, true, options);
+  const { kernelStorage } = store;
+  const artifactMetadata = new Map();
+  for (const [key, value] of exporter.getKVData()) {
+    if (key.startsWith('export.')) {
+      artifactMetadata.set(key, value);
+    } else if (value === null) {
+      kernelStorage.kvStore.delete(key);
+    } else {
+      kernelStorage.kvStore.set(key, value);
+    }
+  }
+  for await (const artifactName of exporter.getArtifactNames(
+    includeHistorical,
+  )) {
+    let fetchedP;
+    if (artifactName.startsWith('export.snapshot.')) {
+      fetchedP = kernelStorage.snapStore.importSnapshot(
+        artifactName,
+        exporter,
+        artifactMetadata,
+      );
+    } else if (artifactName.startsWith('export.transcript.')) {
+      fetchedP = kernelStorage.transcriptStore.importTranscript(
+        artifactName,
+        exporter,
+        artifactMetadata,
+      );
+    } else {
+      Fail`unknown artifact type: ${artifactName}`;
+    }
+    await fetchedP;
+  }
+  return store;
```

Comment:
To avoid burying the data consistency issue raised in https://github.com/Agoric/agoric-sdk/pull/7026#r1111225258, I'm starting a separate thread on the topic of importing over existing data in the future to support rollback.

> > There is of course the question of import onto an existing DB and artifacts already existing in it, but right now we don't seem to support that (we may want to in the future to support rolling back the DB when cosmos does).

@warner in https://github.com/Agoric/agoric-sdk/pull/7026#discussion_r1112462681:
> I think in that case we delete the entire swingstore and perform a new import. Merging an existing DB with an import of some other DB sounds like confusion waiting to happen.

I am concerned about the disk usage of that approach. To copy the heap snapshots and transcripts with that approach, you effectively will be duplicating them during the import, before you can turn around an delete the old DB.

I don't see why we couldn't clear and re-import the kvStore section, trim active transcript spans as necessary, and re-validate that all active heap snapshots (and active transcript span if we aggressively prune those) are available.

Reply:
Can we defer this rollback case for now? For the next two weeks, I think we're talking about one primary use case and one maybe-we-need-it-maybe-not secondary use case:

1: cosmos state-sync, minimal data, no historical data (goal is smallest snapshot possible, smallest resulting new-swingstore size, call it a "pruned swingstore")
2: collecting enough artifacts to repopulate a pruned swingstore, and/or tools to do the repopulation

What I proposed to @FUDCo was:

* exporter only produces minimal exports (no historical data)
* importer adds all artifacts that it receives (after validating them, of course) and aborts if it doesn't get the minimum
* maybe have a flag on the export that produces a complete export, instead of a minimal one
  * especially if it helps us test the importer

Which I think will be enough for the immediate milestone.